ClrMD.Extensions/ClrDynamic.cs

```diff
@@ -670,10 +677,92 @@ public static object GetSimpleValue(ClrDynamic obj)
                 ClrType type = obj.Type;
                 ClrHeap heap = type.Heap;
 
-                if (type.IsPrimitive || type.IsString)
-                    return type.GetValue(obj.Address);
+                ulong unboxedAddress = obj.Address + (ulong)heap.Runtime.DataTarget.DataReader.PointerSize;
+                
+                switch (type.ElementType)
+                {
+                    case ClrElementType.String:
+                    {
+                        return heap.GetObject(obj.Address).AsString();
+                    }
+
+                    case ClrElementType.Boolean:
+                    {
+                        return heap.Runtime.DataTarget.DataReader.Read<bool>(unboxedAddress);
+                    }
+
+                    case ClrElementType.Int8:
+                    {
+                        return heap.Runtime.DataTarget.DataReader.Read<byte>(unboxedAddress);
+                    }
+
+                    case ClrElementType.Int16:
+                    {
+                        return heap.Runtime.DataTarget.DataReader.Read<short>(unboxedAddress);
+                    }
+
+                    case ClrElementType.Int32:
+                    {
+                        return heap.Runtime.DataTarget.DataReader.Read<int>(unboxedAddress);
+                    }
+
+                    case ClrElementType.Int64:
+                    {
+                        return heap.Runtime.DataTarget.DataReader.Read<long>(unboxedAddress);
+                    }
+
+                    case ClrElementType.UInt8:
+                    {
+                        return heap.Runtime.DataTarget.DataReader.Read<sbyte>(unboxedAddress);
+                    }
+                    case ClrElementType.UInt16:
+                    {
+                        return heap.Runtime.DataTarget.DataReader.Read<ushort>(unboxedAddress);
+                    }
+
+                    case ClrElementType.UInt32:
+                    {
+                        return heap.Runtime.DataTarget.DataReader.Read<uint>(unboxedAddress);
+                    }
+
+                    case ClrElementType.UInt64:
+                    {
+                        return heap.Runtime.DataTarget.DataReader.Read<ulong>(unboxedAddress);
+                    }
+
+                    case ClrElementType.NativeInt:
+                    {
+                        heap.Runtime.DataTarget.DataReader.ReadPointer(unboxedAddress, out var value);
+                        return (long)value;
```

Comment:
Not sure why the (long) cast is needed here, it looks like this case should be included with the others just below.